### ISSUE TITLE
@oneOf support broken with interfaces

### ISSUE THREAD

**Issue by levrik** [dotansimha/graphql-code-generator#7938]
### Issue workflow progress

_Progress of the issue based on the [Contributor Workflow](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md#a-typical-contributor-workflow)_

- [x] 1. The issue provides a reproduction available on [GitHub](https://github.com/dotansimha/graphql-code-generator-issue-sandbox-template), [Stackblitz](https://stackblitz.com/github/dotansimha/graphql-code-generator-issue-sandbox-template) or [CodeSandbox](https://codesandbox.io/s/github/dotansimha/graphql-code-generator-issue-sandbox-template)

> Make sure to fork this template and run `yarn generate` in the terminal.
>
> Please make sure the Codegen and plugins version under `package.json` matches yours.

- [ ] 2. A failing test has been provided
- [ ] 3. A local solution has been provided
- [ ] 4. A pull request is pending review

---

**Describe the bug**

**To Reproduce**

Here's a sandbox showing the broken output:

https://codesandbox.io/s/kind-proskuriakova-h2i4te?file=/types.ts

1. My GraphQL schema:

```graphql
directive @oneOf on INPUT_OBJECT

input IntFilterInput @oneOf {
  eq: Int
  ne: Int
  gt: Int
  gte: Int
  lt: Int
  lte: Int
  in: [Int]
}
```

2. My `codegen.yml` config file:

```yaml
schema: schema.graphql
generates:
  types.ts:
    config:
      declarationKind: interface
    plugins:
      - typescript
```

**Expected behavior**

To don't generate invalid TypeScript code.

More specifically either override the `interface` setting and use `type` or fall back to no special `@oneOf` handling.

**Environment:**

- `@graphql-codegen/typescript`: 2.4.2

**Comment by levrik** [comment]
@n1ru4l Since I'm quite busy with other things right now, I hope the linked CodeSandbox above is enough for now.
I'll see what I can do if I have a free minute.

**Comment by levrik** [comment]
I've suggested a fix in #8586
